src/main/kotlin/net/axay/hglaborlobby/gui/guis/PetGUI.kt

```diff
@@ -0,0 +1,124 @@
+        }
+
+        fun spawn(owner: Player) {
+            val currentPetEntity = owner.world.spawnEntity(owner.location, entity)
+            petEntity = currentPetEntity
+            currentPetEntity.isInvulnerable = true
+            currentPetEntity.customName = "${KColors.CORNFLOWERBLUE}${owner.name}'s $name"
+            currentPetEntity.isCustomNameVisible = true
+            if (currentPetEntity is Mob) {
+                val craftMonster = (currentPetEntity as CraftEntity).handle as EntityInsentient
+                clearPathfinders(craftMonster)
+                currentPetEntity.target = null
+                craftMonster.goalSelector.a(0, LaborPathfinderMoveToPlayer(owner, craftMonster))
+                craftMonster.goalSelector.a(1, PathfinderGoalFloat(craftMonster));
+            }
+            if (currentPetEntity is AbstractHorse) {
+                currentPetEntity.addPassenger(owner)
+            }
+            if (currentPetEntity is Tameable) {
+                currentPetEntity.owner = owner
+                currentPetEntity.isTamed = true
+            }
+            if (currentPetEntity is Llama) {
+                currentPetEntity.color = Llama.Color.values().random()
+            }
+            if (currentPetEntity is Wolf) {
+                currentPetEntity.collarColor = DyeColor.values().random()
+            }
+        }
+        private fun clearPathfinders(entity: EntityInsentient) {
+            entity.goalSelector = PathfinderGoalSelector(entity.getWorld().methodProfilerSupplier)
+            entity.targetSelector = PathfinderGoalSelector(entity.getWorld().methodProfilerSupplier)
+        }
+    }
+
+    var Player.pet: Pet?
+        get() = pets[this]
+        set(value) {
+            pets[this] = value!!
+        }
+
+    private val pets = hashMapOf<Player, Pet>()
+
+    private val petsGui = kSpigotGUI(GUIType.THREE_BY_NINE) {
+        title = "${KColors.BLACK}PETS"
+        page(1) {
+            placeholder(Slots.Border, itemStack(Material.WHITE_STAINED_GLASS_PANE) { meta { name = null } })
+            val petCompound = createRectCompound<Pet>(
+                Slots.RowTwoSlotTwo,
+                Slots.RowTwoSlotEight,
+                iconGenerator = {
+                    itemStack(it.icon) {
+                        setMeta {
+                            name = "${KColors.CORAL}${it.name}"
+                            addLore {
+                                +"${KColors.BISQUE}Hole dir ein ${it.name} als treuen Begleiter,"
+                                +"${KColors.BISQUE}um dich in der Lobby aufzuhalten!"
+                                +" "
+                                +"${KColors.LIGHTSLATEGRAY}${KColors.ITALIC}Klicke auf dieses Item,"
+                                +"${KColors.LIGHTSLATEGRAY}${KColors.ITALIC}um das Pet zu beschwören."
+                            }
+                        }
+                    }
+                },
+                onClick = { clickEvent, element ->
```

Comment:
please surround everything inside of the onClick function in a try catch, catching the `NoClassDefFoundError`, because the pathfinder logic is based on version dependent nms imports

Reply:
i just surrounded the nms part to keep the rest running if the server updates to 1.17 or higher